File: src/23-challenge/solution.py
def find_missing_numbers(nums: list[int]) -> list[int]:
    unique_nums: set[int] = set(nums)
    max_unique_num: int = max(unique_nums)

    collection: set[int] = set()
    for i in range(1, max_unique_num):
        collection.add(i)

    missing_nums: list[int] = [*collection.difference(unique_nums)]

    return missing_nums

File: src/23-challenge/solution.ts
export default function findMissingNumbers(nums: number[]): number[] {
	const uniqueNums: Set<number> = new Set(nums)
	const maxUniqueNum: number = Math.max(...uniqueNums)

	const collection: Set<number> = new Set()
	for (let i = 1; i < maxUniqueNum; i++) collection.add(i)

	const missingNums: number[] = [...collection.difference(uniqueNums)]

	return missingNums
}
